src/asmcnc/comms/serial_connection.py

```diff
@@ -380,7 +380,18 @@ def grbl_scanner(self, run_grbl_scanner_once = False):
     
                 # Job streaming: stuff buffer
                 if (self.is_job_streaming and not self.m.is_machine_paused and not "Alarm" in self.m.state()):
-                    if self.yp.use_yp: self.yp.add_to_stack()
+                    if self.yp.use_yp: 
+
+                        if self.digital_spindle_ld_qdA >= 0 \
+                                and self.feed_override_percentage is not None \
+                                and self.feed_rate is not None \
+                                and self.grbl_ln is not None \
+                                and self.digital_spindle_mains_voltage >= 0 \
+                                and self.inrush_counter == 5:
```

Comment:
Inrush counter goes upto 6 because it still adds an extra one after it passes <=5 ;)

Reply:
amended